From 2a46282cac52d3a8ac4e1c3d9a977cde5b853bc4 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 07 Mar 2022 15:01:55 +0200
Subject: [PATCH] [SW-70307] gaudi3: split preboot binary to iccm and dccm

in PLDM simulator there is a requirement to run the preboot
application from iccm/dccm memories. This patch is adding the
a build flag that is propogated to zephyr and also splitting
the preboot elf file into iccm and dccm binaries.

Change-Id: I7997d7818ba2cdffa39f2263c4b71bae097f88bb
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/board/habanalabs/gaudi3/post-image-fit.sh b/board/habanalabs/gaudi3/post-image-fit.sh
index 3777a5a..dd74125 100644
--- a/board/habanalabs/gaudi3/post-image-fit.sh
+++ b/board/habanalabs/gaudi3/post-image-fit.sh
@@ -1,11 +1,20 @@
 #!/usr/bin/env bash
 
+set -e          # stop on any error in this script
+set -o pipefail # stop on any error in pipeline
+
+IFS=$'\n'
+
 ARC_MGMT_DCCM_HI=0x0300007F
 ARC_MGMT_DCCM_LO=0xD3080000
 ARC_MGMT_ICCM_HI=0x0300007F
 ARC_MGMT_ICCM_LO=0xD3040000
 
 HL_ARCH="arc"
+
+OBJCOPY=arc-linux-objcopy
+
+source ${COMMON_BASE}/image-utility.sh
 
 function hl_copy_keys
 {
@@ -18,10 +27,18 @@
 
 function hl_copy_binaries {
 	FILEPATH=$1
+	rodata_iccm=0
+
+	pushd ${BINARIES_DIR}
+
+	# create zephyr iccm and dccm binaries
+	# hl_create_zephyr_binaries "mgmt" ${rodata_iccm}
 
 	# copy binaries from other HL targets required for boot fit creation
 	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin.le
 	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin.le
 
+	popd
+
 	return
 }
